### PR TITLE
Change schema registry tests

### PR DESCRIPTION
**Pull request by RafalKorepta**
To better understand schema registry tests:

* do not `set -x` to not print credentials with schema registry test
* add trap function to report everything about schema registry
* add logs to better trace one of the schema registry test executions
* make curl display information on stdout after a completed transfer - https://curl.se/docs/manpage.html#-w
* Generate unique test to track `ct` execution
* Randomize schema registry subject name

### CODE REVIEW

File: charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml [hunk 1]
```diff
@@ -19,10 +19,11 @@ limitations under the License.
   {{- $cert := get .Values.tls.certs $service.tls.cert -}}
   {{- $root := deepCopy . }}
   {{- $sasl := .Values.auth.sasl }}
+  {{- $randNumber := randNumeric 3 }}
 apiVersion: v1
 kind: Pod
 metadata:
-  name: {{ include "redpanda.fullname" . }}-test-schemaregistry-internal-tls-status
+  name: {{ include "redpanda.fullname" . }}-test-sr-internal-tls-status-{{ $randNumber }}
```
Comment (alejandroEsc) [inline charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml:26]: was this causing collissions? the hooks should have removed this on installs.

File: charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml [hunk 2]
```diff
@@ -58,10 +59,48 @@ spec:
...
+            echo Retrieving subjects/sensor-{{ $randNumber }}-value/versions/latest/schema?deleted=true
+            schemaCurlIgnore https://{{ include "redpanda.internal.domain" . }}:{{ .Values.listeners.schemaRegistry.port }}/subjects/sensor-{{ $randNumber }}-value/versions/latest/schema?deleted=true
+            echo
+          }
+
+{{- $curlOptions := " -svm3 -w \"\\nstatus=%{http_code} %{redirect_url} size=%{size_download} time=%{time_total} content-type=\\\"%{content_type}\\\"\\n\" "}}
+          schemaCurlIgnore () {
+            curl {{ $curlOptions }} --fail --retry "120" --retry-max-time "120" --retry-all-errors -o - \
+              {{- if or (include "sasl-enabled" .|fromJson).bool .Values.listeners.schemaRegistry.authenticationMethod }}
+              -u "${RPK_USER}:${RPK_PASS}" \
```
Comment (alejandroEsc) [inline charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml:94]: useful

File: charts/redpanda/templates/tests/test-schemaregistry-status.yaml [hunk 3]
```diff
@@ -55,37 +56,78 @@ spec:
...
+            echo Retrieving subjects/sensor-{{ $randNumber }}-value/versions/latest?deleted=true
+            schemaCurlIgnore http://{{ include "redpanda.internal.domain" . }}:{{ .Values.listeners.schemaRegistry.port }}/subjects/sensor-{{ $randNumber }}-value/versions/latest?deleted=true
+            echo Retrieving subjects/sensor-{{ $randNumber }}-value/versions/latest/schema
+            schemaCurlIgnore http://{{ include "redpanda.internal.domain" . }}:{{ .Values.listeners.schemaRegistry.port }}/subjects/sensor-{{ $randNumber }}-value/versions/latest/schema
+            echo Retrieving subjects/sensor-{{ $randNumber }}-value/versions/latest/schema?deleted=true
+            schemaCurlIgnore http://{{ include "redpanda.internal.domain" . }}:{{ .Values.listeners.schemaRegistry.port }}/subjects/sensor-{{ $randNumber }}-value/versions/latest/schema?deleted=true
+            echo
+          }
+
+{{- $curlOptions := " -svm3 -w \"\\nstatus=%{http_code} %{redirect_url} size=%{size_download} time=%{time_total} content-type=\\\"%{content_type}\\\"\\n\" "}}
```
Comment (alejandroEsc) [inline charts/redpanda/templates/tests/test-schemaregistry-status.yaml:87]: very nice.

**Review by alejandroEsc — APPROVED** [review]
Just a question

File: charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml (hunk 1 above)
Comment (RafalKorepta) [reply charts/redpanda/templates/tests/test-schemaregistry-internal-tls-status.yaml:26]: I had problem with showing in the following scenario

1. helm install
2. helm test
3. helm upgrade
4. helm test

Schema registry test in point 2 execution is important in the bug report. 

If pod could be changed to Job, maybe I could have both execution point 2 and 4.

**Pull request merged**